rhp/v4/encoding.go

```diff
@@ -0,0 +1,699 @@
+}
+
+func (r *RPCRefreshContractSecondResponse) encodeTo(e *types.Encoder) {
+	r.RenterRenewalSignature.EncodeTo(e)
+	types.EncodeSlice(e, r.RenterSatisfiedPolicies)
+}
+func (r *RPCRefreshContractSecondResponse) decodeFrom(d *types.Decoder) {
+	r.RenterRenewalSignature.DecodeFrom(d)
+	types.DecodeSlice(d, &r.RenterSatisfiedPolicies)
+}
+func (r *RPCRefreshContractSecondResponse) maxLen() int {
+	return reasonableObjectSize
+}
+
+func (r *RPCRefreshContractThirdResponse) encodeTo(e *types.Encoder) {
+	r.Basis.EncodeTo(e)
+	types.EncodeSlice(e, r.TransactionSet)
+}
+func (r *RPCRefreshContractThirdResponse) decodeFrom(d *types.Decoder) {
+	r.Basis.DecodeFrom(d)
+	types.DecodeSlice(d, &r.TransactionSet)
+}
+func (r *RPCRefreshContractThirdResponse) maxLen() int {
+	return reasonableObjectSize
+}
+
+func (r *RPCModifySectorsRequest) encodeTo(e *types.Encoder) {
+	r.ContractID.EncodeTo(e)
+	r.Prices.EncodeTo(e)
+	types.EncodeSlice(e, r.Actions)
+	r.ChallengeSignature.EncodeTo(e)
+}
+func (r *RPCModifySectorsRequest) decodeFrom(d *types.Decoder) {
+	r.ContractID.DecodeFrom(d)
+	r.Prices.DecodeFrom(d)
+	types.DecodeSlice(d, &r.Actions)
+	r.ChallengeSignature.DecodeFrom(d)
+}
+func (r *RPCModifySectorsRequest) maxLen() int {
+	return reasonableObjectSize
+}
+
+func (r *RPCModifySectorsResponse) encodeTo(e *types.Encoder) {
+	types.EncodeSlice(e, r.Proof)
+}
+func (r *RPCModifySectorsResponse) decodeFrom(d *types.Decoder) {
+	types.DecodeSlice(d, &r.Proof)
+}
+func (r *RPCModifySectorsResponse) maxLen() int {
+	return reasonableObjectSize
+}
+
+func (r *RPCModifySectorsSecondResponse) encodeTo(e *types.Encoder) {
+	r.RenterSignature.EncodeTo(e)
+}
+func (r *RPCModifySectorsSecondResponse) decodeFrom(d *types.Decoder) {
+	r.RenterSignature.DecodeFrom(d)
+}
+func (r *RPCModifySectorsSecondResponse) maxLen() int {
+	return sizeofSignature
+}
+
+func (r *RPCModifySectorsThirdResponse) encodeTo(e *types.Encoder) {
+	r.HostSignature.EncodeTo(e)
+}
+func (r *RPCModifySectorsThirdResponse) decodeFrom(d *types.Decoder) {
+	r.HostSignature.DecodeFrom(d)
+}
+func (r *RPCModifySectorsThirdResponse) maxLen() int {
+	return sizeofSignature
+}
+
+func (r *RPCAppendSectorsRequest) encodeTo(e *types.Encoder) {
+	r.Prices.EncodeTo(e)
+	types.EncodeSlice(e, r.Sectors)
+	r.ContractID.EncodeTo(e)
+	r.ChallengeSignature.EncodeTo(e)
+}
+func (r *RPCAppendSectorsRequest) decodeFrom(d *types.Decoder) {
+	r.Prices.DecodeFrom(d)
+	types.DecodeSlice(d, &r.Sectors)
+	r.ContractID.DecodeFrom(d)
+	r.ChallengeSignature.DecodeFrom(d)
+}
+func (r *RPCAppendSectorsRequest) maxLen() int {
+	return reasonableObjectSize
+}
+
+func (r *RPCAppendSectorsResponse) encodeTo(e *types.Encoder) {
+	e.WriteUint64(uint64(len(r.Accepted)))
+
+	var flags uint64
+	for i, accepted := range r.Accepted {
+		if accepted {
+			flags |= uint64(1) << (i % 64)
+		}
+		if (i+1)%64 == 0 {
+			e.WriteUint64(flags)
+			flags = 0
+		}
+	}
+	if len(r.Accepted)%64 != 0 {
+		e.WriteUint64(flags)
+	}
```

Comment:
fun, but surely overkill; if we use the standard 1-byte-per-bool encoding, a 1000-sector append still easily fits in one network packet.

Reply:
Fun removed